source/MissionAction.h

```diff
@@ -66,7 +66,8 @@ class MissionAction {
 	// "Instantiate" this action by filling in the wildcard text for the actual
 	// destination, payment, cargo, etc.
 	MissionAction Instantiate(std::map<std::string, std::string> &subs,
-		const System *origin, int jumps, int64_t payload) const;
+		const System *origin, int jumps, int64_t payload,
+		const std::map<const Outfit *, int> &outfitObjective = {}) const;
```

Comment:
Initiating a vector by default here feels wrong to me

Reply:
why?

Reply:
Lets leave this one for last once other things have been understood and I, or smbd else, will have found a better way (hopefully)

Reply:
I am initializing the map via aggregate initialization so that outfitObjective can be optional. Its just a short hand for std::map<const Outfit *, int>().

Reply:
IE it will create the empty map in memory everytime, which even if empty takes up some memory?

Reply:
Would overloading work better here instead of defaulting?

Reply:
idk I think we need another opinion honestly, not sure whats best in this case either

Reply:
I could add a addGiftCargo method with a signature of (outfit, int) onto GameAction and then a delegation method to MissionAction and then make it a follow up call instead of a smuggled optional argument.

Reply:
This is now done via a AddOutfitObjective method that can be called after Instantiate. Resolving this conversation.